1-Vanilla/main.js

```diff
@@ -0,0 +1,231 @@
+    });
+  };
+  
+  const addTodo = () => {
+    const inputText = document.getElementById("addTodo").value;
+    list.push({
+      text: inputText,
+      status: "unfinished",
+      id: list.length + 1,
+    });
+    loadToDoList(list);
+  };
+  
+  const removeTodo = (list, item) => {
+    list = list.filter(function (ele) {
+      return ele.id != item.id;
+    });
+  
+    loadToDoList(list);
+  };
+  
+  const editTodo = (list, item) => {
+    list = list.map((ele) => {
+      if (ele.id === item.id) {
+        ele.isEditMode = !ele.isEditMode;
+      }
+      return ele;
+    });
+    loadToDoList(list);
+  };
+  
+  const toggleTodo = (item) => {
+    const li = document.getElementById(item.id);
+    const text = li.children[0].children[1];
+    if (item.status === "finished") {
+      text.style["text-decoration"] = "none";
+      item.status = "unfinished";
+    } else {
+      text.style["text-decoration"] = "line-through";
+      item.status = "finished";
+    }
+  };
+  
+  const createButton = (cb, icon, className) => {
+    const button = document.createElement("button");
+    button.innerHTML = `<i class="${icon}" aria-hidden="true"></i>`;
+    button.classList.add(className);
+    button.onclick = cb;
+    button.style.cursor = "pointer";
+    return button;
+  };
+  
+  const createItem = (list, item) => {
+    const li = document.createElement("li");
+    li.id = item.id;
+  
+    const checkbox = document.createElement("input");
+    checkbox.type = "checkbox";
+    checkbox.onclick = () => toggleTodo(item);
+  
+    const spanText = document.createElement("span");
+    spanText.classList.add("spanText");
+    spanText.appendChild(checkbox);
+  
+    if (item.isEditMode) {
+      //edit model
+      const inputText = document.createElement("input");
+      inputText.type = "text";
+      inputText.value = item.text;
+      inputText.classList.add("edit-todo");
+      inputText.id = `${item.id}-editTodo`;
+      spanText.appendChild(inputText);
+    } else {
+      //text mode
+      const text = document.createElement("span");
+      text.innerText = item.text;
+      text.style.marginLeft = "20px";
+      if (item.status === "finished") {
+        text.style["text-decoration"] = "line-through";
+        checkbox.checked = true;
+      }
+      spanText.appendChild(text);
+    }
+  
+    const spanButton = document.createElement("span");
+    spanButton.classList.add("spanActions");
+  
+    const removeButton = createButton(
+      () => removeTodo(list, item),
+      "fa fa-trash",
+      "button"
+    );
+    const editButton = createButton(
+      () => editTodo(list, item),
+      "fa fa-pencil",
+      "edit"
+    );
+    spanButton.appendChild(removeButton);
+    spanButton.appendChild(editButton);
+  
+    li.appendChild(spanText);
+    li.appendChild(spanButton);
+    return li;
+  };
+  
+  const editListenerHandler = () => {
+    const text = e.target.parentNode.children[1];
```

Comment:
I assume you meant to pass `e` as an argument, don't have your code based on the dom structure. it makes the code very fragile and coupled with the html implementation